### PR TITLE
Add links on resource collection

### PR DESCRIPTION
**Pull request by KondukterCRO**

### CODE REVIEW

**Review by iskelic — APPROVED** [review]
Zanemari @KondukterCRO , fulala sam ceo fudbal

File: src/Http/Model/Response/CreatedResponseCollection.php [hunk 1]
```diff
@@ -9,7 +9,7 @@
 use Undabot\JsonApi\Definition\Model\Resource\ResourceCollectionInterface;
 use Undabot\JsonApi\Definition\Model\Resource\ResourceInterface;
 
-final class ResourceCreatedResponse
+final class CreatedResponseCollection implements ResourceResponseInterface
```
Comment (KondukterCRO) [inline src/Http/Model/Response/CreatedResponseCollection.php]: Is this name change required ? I'm thinking of users that used this inside their projects 🤔

File: src/Http/Model/Response/CollectionResourceResponse.php [hunk 2]
```diff
@@ -15,7 +15,7 @@
 use Undabot\JsonApi\Implementation\Model\Resource\ResourceCollection;
 use Undabot\SymfonyJsonApi\Model\Collection\ObjectCollection;
 
-final class ResourceCollectionResponse
+final class CollectionResourceResponse implements ResourceResponseInterface
 {
     /** @var ResourceCollectionInterface */
     private $primaryResources;
```
Comment (KondukterCRO) [inline src/Http/Model/Response/CollectionResourceResponse.php]: Can we add typehint on this property and remove annotation ?

File: src/Http/Model/Response/CreatedResponseCollection.php [hunk 3]
```diff
@@ -9,7 +9,7 @@
...
 {
     /** @var ResourceInterface */
     private $primaryResource;
```
Comment (KondukterCRO) [inline src/Http/Model/Response/CreatedResponseCollection.php]: Maybe add typehint here also ?

File: src/Http/Model/Response/ResourceUpdatedResponse.php [hunk 4]
```diff
@@ -9,7 +9,7 @@
 use Undabot\JsonApi\Definition\Model\Resource\ResourceCollectionInterface;
 use Undabot\JsonApi\Definition\Model\Resource\ResourceInterface;
 
-final class ResourceUpdatedResponse
+final class ResourceUpdatedResponse implements ResourceResponseInterface
 {
     /** @var ResourceInterface */
     private $primaryResource;
```
Comment (KondukterCRO) [inline src/Http/Model/Response/ResourceUpdatedResponse.php]: Here also typehint ?

File: src/Http/Model/Response/ResourceResponse.php [hunk 5]
```diff
@@ -9,7 +9,7 @@
 use Undabot\JsonApi\Definition\Model\Resource\ResourceCollectionInterface;
 use Undabot\JsonApi\Definition\Model\Resource\ResourceInterface;
 
-final class ResourceResponse
+final class ResourceResponse implements ResourceResponseInterface
 {
     /** @var null|ResourceInterface */
     private $primaryResource;
```
Comment (KondukterCRO) [inline src/Http/Model/Response/ResourceResponse.php]: Typehint ?

File: src/Http/Service/EventSubscriber/ViewResponseSubscriber.php [hunk 6]
```diff
@@ -41,21 +44,28 @@ public function buildView(ViewEvent $event): void
     {
         $data = $event->getControllerResult();
 
-        if ($data instanceof ResourceCollectionResponse) {
+        if ($data instanceof CollectionResourceResponse) {
+            $responseLinkBuilder = new ResponseLinksBuilder(
```
Comment (KondukterCRO) [inline src/Http/Service/EventSubscriber/ViewResponseSubscriber.php]: Why not to pull this from container ?

Comment (iskelic) [reply src/Http/Service/EventSubscriber/ViewResponseSubscriber.php]: Because laziness? :D Good idea, thanks for the tip

**Comment by ghost** [comment]
* [x] added named constructors to PaginationModel
* [x] expanded URL builders with abs URL instead of rel
* [x] finished some unit/integration tests regarding URL building

**Pull request closed without merging**